dist/gcc-compatible/Hacl_Hash_Blake2.c

```diff
@@ -500,16 +501,72 @@ void Hacl_Blake2b_32_blake2b_init(uint64_t *hash, uint32_t kk, uint32_t nn)
   r3[1U] = iv5;
   r3[2U] = iv6;
   r3[3U] = iv7;
-  uint64_t kk_shift_8 = (uint64_t)kk << (uint32_t)8U;
-  uint64_t iv0_ = iv0 ^ ((uint64_t)0x01010000U ^ (kk_shift_8 ^ (uint64_t)nn));
+  uint8_t salt[16U] = { 0U };
+  uint8_t personal[16U] = { 0U };
+  Hacl_Impl_Blake2_Core_blake2b_params
```

Comment:
can this be inlined for extraction?

Reply:
maybe [@inline_let]

Reply:
This one is a bit unfortunate, but all the functions/stack allocations are in the stack effect, including packing all elements in the params struct (to ensure that the struct satisfies some invariants after initialization, i.e., the salt and personal arrays are live). Hence, inline_let does not work here.